components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -1725,7 +1725,13 @@ public static String getPASTRCookieName (String sessionDataKey) {
                 idpMappedUserRoles.add(idpRole);
             }
         }
-        return idpMappedUserRoles;
+        List<String> idpMappedUniqueUserRoles = new ArrayList<>();
+        for (String role: idpMappedUserRoles) {
+            if (!idpMappedUniqueUserRoles.contains(role)) {
+                idpMappedUniqueUserRoles.add(role);
+            }
+        }
+        return idpMappedUniqueUserRoles;
```

Comment:
As the expectation of this is to return the local roles mapped it would better to use a set rather than re-populating an arraylist object

Reply:
I will fix this